squire_sdk/src/server/gathering/hall.rs

```diff
@@ -85,7 +85,9 @@ where
         match msg {
             GatheringHallMessage::NewGathering(id) => self.process_new_gathering(id).await,
             GatheringHallMessage::NewConnection(id, user, ws) => {
-                self.process_new_onlooker(id, user, ws).await
+                if !self.process_new_onlooker(id, user, ws).await {
+                    panic!("process_new_onlooker failed")
+                }
```

Comment:
So now I have to decide on these panics I added. I think the right call is context specific. But I will say that if ignore the error instead, it won't be a regression (and as such maybe we can handle it in a separate PR). The question is about whether to handle a failure to send now that we have access to it (`bool` return value of `send()`), and whether `panic` is the right option.

`process_new_onlooker` gets or inits a gathering. Supposing we get one and it fails to send. Rather than panic, it seems like maybe what needs to happen is that these gatherings should be removed from the collection? Since they've panicked or shut down?

Reply:
> But I will say that if ignore the error instead, it won't be a regression
This is correct. This function returns `false` when the user attempts to connect to a tournament that has already shutdown for whatever reason. This is not an unrecoverable error for the `GatheringHall`. A panic here means, amount other things, that **all** tournaments will fail to get new connections.

I think the best solution for now is to just ignore this error. I'm working to refactor most of the websocket management logic so that a user only ever has one WS open at a time. That will address most of the open questions around this problem.